charmcraft/linters.py

```diff
@@ -67,6 +68,27 @@ def check_dispatch_with_python_entrypoint(basedir: pathlib.Path) -> pathlib.Path
     return None
 
 
+def check_naming_convention(names: typing.Iterable[str], scope: str) -> tuple[str, str]:
+    snake_keys = [key for key in names if "_" in key]
+
+    if snake_keys:
+        hyphen_keys = [key for key in names if "-" in key]
```

Comment:
@lengau is there a clear preference for one or the other? If we're warning against mixed use, should we also steer towards the preferred one?

Reply:
@cmatsuoka The preferred way is `-` following the [styleguide](https://juju.is/docs/sdk/styleguide#heading--naming)